server/src/main/java/org/candlepin/dto/api/v1/EntitlementDTO.java

```diff
@@ -0,0 +1,541 @@
+                .append(thisPoolId, thatPoolId)
+                .append(thisConsumerId, thatConsumerId)
+                .append(this.getQuantity(), that.getQuantity())
+                .append(this.isDirty(), that.isDirty())
+                .append(this.getEndDateOverride(), that.getEndDateOverride())
+                .append(this.isUpdatedOnStart(), that.isUpdatedOnStart())
+                .append(this.isDeletedFromPool(), that.isDeletedFromPool())
+                .append(this.getEndDate(), that.getEndDate())
+                .append(this.getStartDate(), that.getStartDate());
+
+            // As with many collections here, we need to explicitly check the elements ourselves,
+            // since it seems very common for collection implementations to not properly implement
+            // .equals
+            // Note that we're using the boolean operator here as a shorthand way to skip checks
+            // when the equality check has already failed.
+            boolean equals = builder.isEquals();
+
+            equals = equals && Util.collectionsAreEqual(this.getCertificates(), that.getCertificates());
+
+            return equals;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        int certsHashCode = 0;
+        Collection<CertificateDTO> certificateDTOs = this.getCertificates();
+
+        if (certificateDTOs != null) {
+            for (CertificateDTO dto : certificateDTOs) {
+                certsHashCode = 31 * certsHashCode + (dto != null ? dto.hashCode() : 0);
+            }
+        }
+
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+            .append(super.hashCode())
+            .append(this.getId())
+            .append(this.getOwner() != null ? this.getOwner().getId() : null)
+            .append(this.getPool() != null ? this.getPool().getId() : null)
+            .append(this.getConsumer() != null ? this.getConsumer().getUuid() : null)
+            .append(this.getQuantity())
+            .append(this.isDirty())
+            .append(this.getEndDateOverride())
+            .append(this.isUpdatedOnStart())
+            .append(this.isDeletedFromPool())
+            .append(this.getEndDate())
+            .append(this.getStartDate())
+            .append(certsHashCode);
+
+        return builder.toHashCode();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EntitlementDTO clone() {
+        EntitlementDTO copy = (EntitlementDTO) super.clone();
+
+        OwnerDTO owner = this.getOwner();
+        copy.owner = owner != null ? owner.clone() : null;
+        PoolDTO pool = this.getPool();
+        copy.pool = pool != null ? pool.clone() : null;
+        Consumer consumer = this.getConsumer();
+        //TODO change Consumer to ConsumerDTO.
+        copy.consumer = consumer != null ?
+            new Consumer(consumer.getName(), consumer.getUsername(),
```

Comment:
I definitely would not use new here if it wasn't for the fact that Consumer does not implement clone(), and did not want to introduce it just for the sake of using it here, since this code will get replaced by ConsumerDTO which I expect to have a clone() method.